config/config_params.go

```diff
@@ -317,6 +317,7 @@ func (config *Config) DatastoreConfig() api.CalicoAPIConfig {
 		if err != nil {
```

Comment:
Ah, I see now; and also

	DatastoreType DatastoreType `json:"datastoreType" envconfig:"DATASTORE_TYPE" default:"etcdv2"`

OK, so this PR is wrong then, I'll close it.  Instead, in the longer term, I guess it would be nice if we could remove the need for setting FELIX_DATASTORETYPE as well as DATASTORE_TYPE or CALICO_DATASTORE_TYPE.